### ISSUE TITLE
Error: Connection failed : Cannot resolve host name 

### ISSUE THREAD

**Issue by bkathiry** [aws-samples/amazon-cloudwatch-monitor-for-sap-hana#6]
Hello Marcel , AWS Team,

We have HANA 2.0 database , we tried to implement HANA database monitoring using AWS Lambda but getting an error while connecting to HANA database.
Here are more details of the issue.
We have private DNS server so , we are giving IP of the HANA database and still code is doing DNS resolution and failing as below(attachment - Error_with_IP_address) .
![Error_with_IP_Aaddress](https://user-images.githubusercontent.com/110205568/181653505-e2688fae-8369-4657-8eb5-97adaee74e41.JPG)

If I give AWS internal hostname then also same issue.. but it connects to different port 30015 (attachment - Error_with_AWS_Internalhost)

![Error_with_AWS_internalhost](https://user-images.githubusercontent.com/110205568/181653524-e3b3699e-7da1-4e8f-aa06-600015bf54e2.JPG)

If I give correct DB hostname then also same issue.
![Error_with_DB_Hostname](https://user-images.githubusercontent.com/110205568/181653535-5af057ea-27fa-45a8-af37-559bc25e7850.JPG)

  Once we pass IP address as parameter then it should connect to tenant db port 30013 (instead of 30015 port as above error)directly without any DNS lookup errors.Could you please check and help us to resolve the code or libraries issue ?
Thanks for your support.

**Comment by frumania** [comment]
Hi,
thanks for contacting!
Is /etc/hosts on the HANA OS properly configured to map this DNS with your local private IP?

**Comment by bkathiry** [comment]
Hi Marcel,

Yes, Host name resolution is happening via private DNS server , so we don't maintain IP address resolution in /etc/hosts. 
SAP and DB is working fine with this host name resolution from many months , so I don't believe we have issue there.
Once we connect via IP address , it should not look for Host name resolution..correct ?
We can meet via screen sharing session for more details about the issue , please let me know. Thanks for your response.

**Comment by frumania** [comment]
I agree. My assumption is, that this is simply a feature of the HANA client.
Let's see, happy to have a screen sharing session.
Can you contact me here?!
https://www.linkedin.com/in/marceltoerpe/

**Comment by bkathiry** [comment]
Sure Marcel. Thanks.

**Comment by Xblade007** [comment]
@frumania  We are again facing this issue, can you please help to understand why this is occuring again.

**Comment by frumania** [comment]
I just did a refresh of the HANA Client. Can you recheck, if it by chance resolves the issue?
If not, happy to join a call and try to troubleshoot! Linkedin Contact see above.

**Comment by Xblade007** [comment]
Hi Marcel,

On linkdin, I dont see any contact. Let me know if need to schedule meeting
at my end.

Still same issue, please refer below error.

{
  "errorType": "Error",
  "errorMessage": "Connection failed (RTE:[89009] Socket connect timeout
(60000 ms) (169.254.76.1:58176 -> 190.166.65.90:30013))",
  "trace": [
    "Error: Connection failed (RTE:[89009] Socket connect timeout (60000
ms) (169.254.76.1:58176 ->  190.166.65.90 :30013))",
    "    at Socket.<anonymous>
***@***.***/hana-client/lib/index.js:53:13)",
    "    at Socket.emit (node:events:513:28)",
    "    at Socket.emit (node:domain:489:12)",
    "    at addChunk (node:internal/streams/readable:315:12)",
    "    at readableAddChunk (node:internal/streams/readable:289:9)",
    "    at Socket.Readable.push (node:internal/streams/readable:228:10)",
    "    at Pipe.onStreamRead (node:internal/stream_base_commons:190:23)"
  ]
}

Regards,
Chanchal

**Comment by frumania** [comment]
Yes, let us please schedule a meeting
You contact me here privately:
https://www.linkedin.com/in/marceltoerpe/

**Comment by Xblade007** [comment]
Can we connect on google meet.
##removed## is my email.
